### PR TITLE
add query_start_time_microseconds field to system.query_log & system.query_thread_log tables

### PR DESCRIPTION
**Pull request by bharatnc**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Add `query_start_time_microseconds` field to `system.query_log` &  `system.query_thread_log` tables.

Detailed description / Documentation draft:

Add a new field called `query_start_time_microseconds`  to `system.query_log` 
& `system.query_thread_log`  tables for having `microsecond` precision in addition 
to the regular DateTime format.  This new field will be of type `DateTime64` and should 
give better accuracy and visibility into query timings. 

This is what the changes in this PR look like:

```sql
SELECT 
    query_id,
    query_start_time,
    query_start_time_microseconds
FROM system.query_log
LIMIT 3

┌─query_id─────────────────────────────┬────query_start_time─┬─query_start_time_microseconds─┐
│ df11cd7c-52c7-4e12-ba2f-0e12017d2dcb │ 2020-09-04 10:36:54 │    2020-09-04 10:36:54.809193 │
│ df11cd7c-52c7-4e12-ba2f-0e12017d2dcb │ 2020-09-04 10:36:54 │    2020-09-04 10:36:54.809193 │
│ 5b2b6719-6ba0-480b-8f9f-dbca41c6cb71 │ 2020-09-04 10:36:59 │    2020-09-04 10:36:59.538415 │
└──────────────────────────────────────┴─────────────────────┴───────────────────────────────┘

3 rows in set. Elapsed: 0.018 sec. 

SELECT 
    query_id,
    query_start_time,
    query_start_time_microseconds
FROM system.query_thread_log
LIMIT 3

┌─query_id─────────────────────────────┬────query_start_time─┬─query_start_time_microseconds─┐
│ df11cd7c-52c7-4e12-ba2f-0e12017d2dcb │ 2020-09-04 10:36:54 │    2020-09-04 10:36:54.900885 │
│ df11cd7c-52c7-4e12-ba2f-0e12017d2dcb │ 2020-09-04 10:36:54 │    2020-09-04 10:36:54.899900 │
│ df11cd7c-52c7-4e12-ba2f-0e12017d2dcb │ 2020-09-04 10:36:54 │    2020-09-04 10:36:54.900338 │
└──────────────────────────────────────┴─────────────────────┴───────────────────────────────┘

3 rows in set. Elapsed: 0.018 sec. 
```

This is related to: #9800

### CODE REVIEW

**Comment by bharatnc** [comment]
The one failing test: https://clickhouse-builds.s3.yandex.net/14252/faa06ed68fb6d71458bc4d6bde39357d89cea2e4/clickhouse_build_check/build_log_760136445_1598813833.txt,  is unrelated to this PR and was already broken. Since then, it has been fixed upstream so merged my branch with master to pull in those changes.

File: src/Interpreters/QueryThreadLog.cpp [hunk 1]
```diff
@@ -1,70 +1,69 @@
...
-        {std::make_shared<DataTypeUInt32>(),        "client_version_patch"},
-        {std::make_shared<DataTypeUInt8>(),         "http_method"},
-        {std::make_shared<DataTypeString>(),        "http_user_agent"},
-        {std::make_shared<DataTypeString>(),        "quota_key"},
-
-        {std::make_shared<DataTypeUInt32>(),        "revision"},
+    return {
+        {std::make_shared<DataTypeDate>(), "event_date"},
+        {std::make_shared<DataTypeDateTime>(), "event_time"},
+        {std::make_shared<DataTypeDateTime>(), "query_start_time"},
```
Comment (nikitamikhaylov) [inline src/Interpreters/QueryThreadLog.cpp:26]: Why did you reformat this code?

File: src/Interpreters/QueryLog.cpp [hunk 2]
```diff
@@ -1,88 +1,85 @@
...
-        {std::make_shared<DataTypeUInt32>(),                                  "client_version_major"},
-        {std::make_shared<DataTypeUInt32>(),                                  "client_version_minor"},
-        {std::make_shared<DataTypeUInt32>(),                                  "client_version_patch"},
-        {std::make_shared<DataTypeUInt8>(),                                   "http_method"},
-        {std::make_shared<DataTypeString>(),                                  "http_user_agent"},
-        {std::make_shared<DataTypeString>(),                                  "quota_key"},
-
-        {std::make_shared<DataTypeUInt32>(),                                  "revision"},
+    auto query_status_datatype = std::make_shared<DataTypeEnum8>(DataTypeEnum8::Values{
+        {"QueryStart", static_cast<Int8>(QUERY_START)},
```
Comment (nikitamikhaylov) [inline src/Interpreters/QueryLog.cpp:28]: And this code...

File: src/Interpreters/QueryLog.cpp [hunk 3]
```diff
@@ -1,88 +1,85 @@
...
+#include <DataTypes/DataTypeFactory.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypesNumber.h>
 #include <Interpreters/ProfileEventsExt.h>
+#include <Interpreters/QueryLog.h>
 #include <Poco/Net/IPAddress.h>
-#include <array>
+#include <Common/ClickHouseRevision.h>
+#include <Common/IPv6ToBinary.h>
+#include <Common/ProfileEvents.h>
```
Comment (nikitamikhaylov) [inline src/Interpreters/QueryLog.cpp:20]: This reformatting is Ok.

**Review by nikitamikhaylov — CHANGES_REQUESTED** [review]

**Comment by nikitamikhaylov** [comment]
Query found by fuzzer seems to be already fixed. Please, rebase on master.

File: src/Interpreters/QueryThreadLog.cpp (hunk 1 above)
Comment (bharatnc) [reply src/Interpreters/QueryThreadLog.cpp]: I did a general formatting of the file, will keep in mind to not format blocks like this in future.

**Review by nikitamikhaylov — APPROVED** [review]

**Pull request merged**